src/main/scala/piconot/external/parser/Parser.scala

```diff
@@ -0,0 +1,45 @@
+import scala.util.parsing.combinator._
+import piconot.ir._
+
+
+object PicoParser extends JavaTokenParsers with PackratParsers{
+
+    // parsing interface
+    def apply(s: String): ParseResult[Rule] = parseAll(expr, s)
+
+    // expressions
+    lazy val expr: PackratParser[Rule] = 
+      (   "Label"~name~":"~body ^^ {case "Label"~n~":"~b ⇒ Label(n)(b)}  )
+        
+    // factors
+    lazy val body: PackratParser[Rule] =
+      (   "Face"~direction ^^ {case "Face"~d ⇒ Face(d)}  
+        | "Go until"~surrounding ^^ {case "Go until"~s => Go.until(s)}
+        | "Jump"~expr            ^^ {case "Jump"~e => Jump(e)})
+      
+    lazy val direction: PackratParser[Rule] = 
+      (   "North"   ^^ {case "North" => NORTH}
+        | "West"   ^^ {case "West" => WEST}
+        | "South"   ^^ {case "South" => SOUTH}
+        | "East"   ^^ {case "East" => EAST})
+
+    lazy val surrounding: PackratParser[Rule] = 
+      (   "BACK_WALL"               ^^ {case "BACK_WALL" => BACK_WALL}
```

Comment:
This is concerning. It seems like there should be a much cleaner way to express these ideas